libraries/eosiolib/contracts/include/eosio/multi_index.hpp

```diff
@@ -2314,5 +2326,34 @@ namespace eosio
 
          _items_vector.erase(--(itr2.base()));
       }
+
+      /**
+       * Updates the table for a newly added index.  Updates at most max_steps rows.
+       * next_primary_key should start at 0.
+       */
+      template <eosio::name::raw IndexName>
+      uint32_t update_index(eosio::name payer, uint64_t& next_primary_key, uint32_t max_steps)
+      {
+         using namespace _multi_index_detail;
+
+         using index_type = decltype(get_index<IndexName>());
+         for (auto iter = lower_bound(next_primary_key), end = this->end();
+              max_steps > 0 && iter != end; --max_steps, ++iter)
+         {
+            auto secondary = index_type::extract_secondary_key(*iter);
+            auto existing = secondary_index_db_functions<
+                typename index_type::secondary_key_type>::db_idx_find_primary(_code.value, _scope,
+                                                                              index_type::name(),
+                                                                              iter->primary_key(),
+                                                                              secondary);
+            if (existing < 0)
+            {
+               secondary_index_db_functions<typename index_type::secondary_key_type>::db_idx_store(
+                   _scope, index_type::name(), payer.value, iter->primary_key(), secondary);
+            }
+            next_primary_key = iter->primary_key() + 1;
```

Comment:
wraps if a pk exists that's 0xffff...

Reply:
That isn't possible in the current usage, since the pk of the member table must be an account that exists.